### PR TITLE
Feature/no count

### PR DESCRIPTION
**Pull request by SimonSchick**
https://github.com/WatchBeam/backend/pull/1291#discussion_r112338609

### CODE REVIEW

File: src/Iterator.ts [hunk 1]
```diff
@@ -37,6 +37,9 @@ export class IteratorImplementation<T extends IModel> implements IIterator<T> {
     }
 
     public getTotal (): number {
+        if (this.results.total === -1) {
```
Comment (ProbablePrime) [inline src/Iterator.ts]: Here now we've removed the header probably `if (!this.results.total)` ?

Comment (SimonSchick) [inline src/Iterator.ts]: How silly of me.

**Comment by SimonSchick** [comment]
https://trello.com/c/CrEH0rRx/103-experiment-with-removing-the-x-total-count-header-for-performance

File: src/adapters/components/QueryBuilder.spec.ts [hunk 2]
```diff
@@ -132,6 +132,18 @@ describe('QueryBuilder', () => {
...
 
+    describe('noCount', () => {
+        it('Adds it correctly', () => {
+            expect(builder.build({
+                noCount: true,
+            }).noCount).to.equal(1);
+        });
+
+        it('Is 0 by default', () => {
+            expect(builder.build({}).noCount).to.equal(0);
```
Comment (connor4312) [inline src/adapters/components/QueryBuilder.spec.ts]: Shouldn't nocount be on by default?

Comment (SimonSchick) [inline src/adapters/components/QueryBuilder.spec.ts]: No, as it would be breaking behaviour.

Comment (connor4312) [inline src/adapters/components/QueryBuilder.spec.ts]: Is that an issue? Do we actually use the x-total-count anywhere?

Comment (SimonSchick) [inline src/adapters/components/QueryBuilder.spec.ts]: I think only in the infinite scrollers, but I guess I can deactivate it there.

Comment (SimonSchick) [inline src/adapters/components/QueryBuilder.spec.ts]: Done @connor4312

**Pull request merged**